### DISCUSSION TITLE
Error when pushing to origin.

### DISCUSSION THREAD

**Discussion by makgtrt06** [community/community#37129] [Other Feature Feedback, Questions, & Ideas]
When I try to push to origin, this error pops up:

_The push operation includes a file which exceeds GitHub's file size restriction of 100MB. Please remove the file from history and try again._

How do I remove the file from the history?

**Comment by FatChicken277** [comment]
1. First you must remove the last commit: `git reset --soft HEAD~1` (using --soft option you will not lose your work)
2. Unstage +100MB file: `git rm --cached <file>`
3. Create the commit again
4. Push

If I answered your question, then you should close the question by click accept answer button :heart:

**Reply by queenofcorgis** [reply]
Thanks for jumping in on this so quickly @FatChicken277 ⭐

**Reply by makgtrt06** [reply]
Thank you for your reply, but when I execute `git reset --soft HEAD~1` in the terminal, it says: `fatal: not a git repository (or any of the parent directories): .git`

**Reply by FatChicken277** [reply]
@makgtrt06  It is because you are not located in the repository folder or you have not initialized the repository ("git init")

**Reply by makgtrt06** [reply]
Okay, but now I realised that the commit is an old one, so the second command doesn't work.

**Reply by FatChicken277** [reply]
How old?, you can undo commits x times: `git reset --soft HEAD~x` (Remember that you already did it 1 time)

So for example: 

> git add -A; git commit -m "Start here."
> git add -A; git commit -m "One"
> git add -A; git commit -m "Two"
> git add -A' git commit -m "Three"

so we have 4 commits now
> * da883dc (HEAD, master) Three
> * 92d3eb7 Two
> * c6e82d3 One
> * e1e8042 Start here.

undo last 3 commits
> git reset --soft HEAD~3

> * e1e8042 Start here.

**Comment by ossanna16** [comment]
Hi there @makgtrt06 and welcome to our community! Thank you for asking a great question 🙂
